### PR TITLE
Separating the UCA Model from Credential Commons so that it can be integrated independently

### PR DESCRIPTION
**Pull request by brunocarneiro**
Separating the UCA Model from Credential Commons so that it can be integrated independently

### CODE REVIEW

**Comment by brunocarneiro** [comment]
There're changes. UserCollectableAttributes does not have anything regarding to attestation.
All those things are now on the new object called Claim, that resides on credential-commons and extends UserCollectableAttributes.

**Comment by brunocarneiro** [comment]
Yes, CI is working. Please check here:
https://circleci.com/gh/identity-com/uca/6

There are unit tests, will check if the coverage is running and what is the current coverage percentage.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    "test:watch": "jest --watch",
+    "build:cjs": "BABEL_ENV=cjs babel src -d dist/cjs",
+    "build:es": "babel src -d dist/es --no-babelrc",
+    "build:browser:before": "BABEL_ENV=browser babel src -d dist/prebrowser",
+    "build:browser:after": "minify dist/prebrowser -d dist/browser",
+    "build:browser:clean": "rimraf dist/prebrowser",
+    "build:browser": "npm run build:browser:before && npm run build:browser:after && npm run build:browser:clean",
+    "build": "npm run build:browser && npm run build:cjs && npm run build:es"
+  },
+  "author": "",
```
Comment (dmelosantos) [inline package.json]: Don't we have an author or array of them?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    "eslint-plugin-import": "^2.13.0",
+    "eslint-plugin-jsx-a11y": "^6.1.1",
+    "eslint-plugin-no-only-tests": "^2.0.0",
+    "eslint-plugin-react": "^7.10.0",
+    "husky": "^1.1.2",
+    "jest": "^23.6.0",
+    "jest-html-reporter": "^2.3.0",
+    "request-debug": "^0.2.0",
+    "rimraf": "^2.6.2"
+  }
```
Comment (dmelosantos) [inline package.json]: Coverage and reports of JEST are not enabled, see credential-commons style

**Comment by brunocarneiro** [comment]
85% of tests coverage now. Fixed package.json attributes as well.

File: __test__/UserCollectableAttributes.test.js [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+    function createUCA() {
+      return new UCA(identifier, value);
+    }
+
+    expect(createUCA).toThrow();
+  });
+
+
+  test('UCA has type String', () => {
```
Comment (jpsantosbh) [inline __test__/UserCollectableAttributes.test.js]: cvc:Verify:phoneNumberToken UCA has type string, not all UCAs

File: __test__/UserCollectableAttributes.test.js [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+    const identifier = 'cvc:Verify:phoneNumberToken';
+    const value = '12345';
+    const v = new UCA(identifier, value);
+    expect(v).toBeDefined();
+    expect(v.type).toEqual('String');
+  });
+
+  test('UCA has type Object:cvc:Identity:name', () => {
```
Comment (jpsantosbh) [inline __test__/UserCollectableAttributes.test.js]: same thing

File: __test__/UserCollectableAttributes.test.js [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+    const identifier = 'cvc:Identity:name';
+    const value = {
+      givenNames: 'joao',
+    };
+    const v = new UCA(identifier, value);
+    expect(v).toBeDefined();
+    expect(v.type).toEqual('Object');
+  });
+
+  test('UCA has incorrect object value', () => {
```
Comment (jpsantosbh) [inline __test__/UserCollectableAttributes.test.js]: The test description is not clear

File: __test__/UserCollectableAttributes.test.js [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+    const v = new UCA(identifier, value);
+
+    expect(v).toBeDefined();
+    expect(v.value).toHaveProperty('givenNames');
+    expect(v.value.givenNames.value).toEqual('Joao Paulo');
+    expect(v.value).toHaveProperty('familyNames');
+    expect(v.value.familyNames.value).toEqual('Barbosa Marques dos Santos');
+  });
+
+  test('UCA has incorrect complex object value', () => {
```
Comment (jpsantosbh) [inline __test__/UserCollectableAttributes.test.js]: the test description is not clear

File: __test__/UserCollectableAttributes.test.js [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+    expect(v).toBeDefined();
+  });
+
+  test('Construct NameGivenNames', () => {
+    const v = new UCA.IdentityName({ givenNames: 'Joao', otherNames: 'Barbosa', familyNames: 'Santos' });
+    expect(v).toBeDefined();
+  });
+
+
+  test('UCA return complex/multiple Attestatble Values', () => {
```
Comment (jpsantosbh) [inline __test__/UserCollectableAttributes.test.js]: UCAs don't know about attestable values

File: __test__/UserCollectableAttributes.test.js [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+    const v = new UCA.IdentityName({ givenNames: 'Joao', otherNames: 'Barbosa', familyNames: 'Santos' });
+    expect(v).toBeDefined();
+  });
+
+  test('UCA should Construct with a complex Attestatble Value: cvc:Identity:name', () => {
```
Comment (jpsantosbh) [inline __test__/UserCollectableAttributes.test.js]: Should NOT, just fix the test description

File: package.json [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    "build:browser:after": "minify dist/prebrowser -d dist/browser",
+    "build:browser:clean": "rimraf dist/prebrowser",
+    "build:browser": "npm run build:browser:before && npm run build:browser:after && npm run build:browser:clean",
+    "build": "npm run build:browser && npm run build:cjs && npm run build:es"
+  },
+  "author": "Identity.com Community",
+  "license": "MIT",
+  "dependencies": {
+    "lodash": "^4.17.11",
+    "sjcl": "^1.0.8",
```
Comment (jpsantosbh) [inline package.json]: please use civic fork for SJCL

**Pull request merged**